include/lib/utils_def.h

```diff
@@ -57,16 +57,19 @@
 	(((ptr) > UINTPTR_MAX - (inc)) ? 1 : 0)
 
 /*
- * For those constants to be shared between C and other sources, apply a 'ull'
- * suffix to the argument only in C, to avoid undefined or unintended behaviour.
+ * For those constants to be shared between C and other sources, apply a 'ul'
+ * or 'ull' suffix to the argument only in C, to avoid undefined or unintended
+ * behaviour.
  *
- * The GNU assembler and linker do not support the 'ull' suffix (it causes the
- * build process to fail) therefore the suffix is omitted when used in linker
- * scripts and assembler files.
+ * The GNU assembler and linker do not support the 'ul' and 'ull' suffix (it
+ * causes the build process to fail) therefore the suffix is omitted when used
+ * in linker scripts and assembler files.
 */
 #if defined(__LINKER__) || defined(__ASSEMBLY__)
+# define  UL(_x)	(_x)
 # define ULL(_x)	(_x)
 #else
+# define  UL(_x)	(_x##ul)
```

Comment:
As far as I know, using `ULL()` should be enough because it ensures that the constant is treated as a 64-bit value when compiling for both architectures.